tournamentcontrol/competition/utils.py

```diff
@@ -426,11 +426,20 @@ def stage_group_position(stage, formula):
     if g is None:
         logger.debug("No group specified in formula %r", formula)
     else:
-        try:
-            g = s.pools.all()[int(g) - 1]
-        except IndexError:
-            logger.exception("Invalid group %r for stage %r", g, stage)
-            raise
+        pools = s.pools.all()
+        if pools and len(pools) >= int(g):
```

Comment:
Use `QuerySet.count` rather than `len` function.

The `pools` variable can never be `None` in this refactor, so this conditional can be simplified.